crates/cast/src/lib.rs

```diff
@@ -981,6 +994,107 @@ impl<P: Provider<AnyNetwork>> Cast<P> {
         Ok(res)
     }
 
+    fn extract_block_range(filter: &Filter) -> Result<(Option<u64>, Option<u64>)> {
+        let FilterBlockOption::Range { from_block, to_block } = &filter.block_option else {
+            return Ok((None, None));
+        };
+
+        let extract_number = |block: Option<BlockNumberOrTag>| {
+            block.and_then(|b| match b {
+                BlockNumberOrTag::Number(n) => Some(n),
+                _ => None,
+            })
+        };
+
+        Ok((extract_number(*from_block), extract_number(*to_block)))
```

Comment:
let's also make this a helper on filter type itself by adding a function for this on alloy?